client/driver/executor/executor.go

```diff
@@ -334,6 +350,28 @@ func (e *UniversalExecutor) ShutDown() error {
 	return nil
 }
 
+func (e *UniversalExecutor) RegisterServices() error {
+	e.logger.Printf("[INFO] executor: registering services")
+	if e.consulService == nil {
+		cs, err := consul.NewConsulService(e.ctx.ConsulConfig, e.logger, e.ctx.AllocID)
+		if err != nil {
+			return err
+		}
+		e.consulService = cs
+	}
+	err := e.consulService.SyncTask(e.ctx.Task)
+	go e.consulService.PeriodicSync()
+	return err
+}
+
+func (e *UniversalExecutor) DeregisterServices() error {
+	e.logger.Printf("[ERR] executor: de-registering services and shutting down consul service")
```

Comment:
@dadgar I think having it here helps in understanding whether the method was being called, also should be `INFO` :-/